### PR TITLE
[FL2-6-FL2-452] Create a 'Bar' component

### PR DESCRIPTION
**Pull request by st-build-service**
Fixed and raised by ricardo
capture

### CODE REVIEW

File: components/bars/Bar.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    render () {
+        const color = `${this.props.color || 'blue'}`;
+        const cls = classNames('m-bar', this.props.className, color, {
+            'active': this.props.active
+        });
+
+        const style = {
+            height: this.props.height,
+            width: this.props.width
+        };
```
Comment (jimmed) [inline components/bars/Bar.js]: :point_up: Destructuring is your friend <3

```js
const { className, active, height, width, color = 'blue' } = this.props;
const cls = classNames('m-bar', className, color, { active });
const style = { width, height };
```

File: components/bars/Bar.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+        return (
+            <div
+                data-tip={this.props.tip}
+                className={cls}
+                style={style}
+                onClick={this.handleClick.bind(this)}>
+                {this.props.children}
+                {this.props.tip && (
+                    <Portal isOpened={true}>
```
Comment (jimmed) [inline components/bars/Bar.js]: `<Portal isOpened={true}>` => `<Portal isOpened>`

File: pages/componentsonly.js [hunk 3]
```diff
@@ -709,6 +710,68 @@ class ButtonDemo extends React.Component {
...
+                                height={height}
+                                width={width}
+                                color={color}
+                                tip={tip}
+                                active={active}
+                            >
+                                {content}
+                            </Bar>
+                        )
+                    )
```
Comment (jimmed) [inline pages/componentsonly.js]: For your own sanity:

```js
{barExamples.map(({ content, ...rest }) => (
   <Bar {...rest}>{content}</Bar>
))}
```

File: components/bars/Bar.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
+import React from "react"
+import classNames from "classnames"
+import ReactTooltip from "react-tooltip"
+import Portal from "react-portal"
+
+export default class Bar extends React.PureComponent {
+    static defaultProps = {
```
Comment (victorhtc) [inline components/bars/Bar.js]: I'm missing the propTypes.

**Pull request merged**